src/Agent.Listener/SelfUpdater.cs

```diff
@@ -128,7 +128,25 @@ private async Task<bool> UpdateNeeded(string targetVersion, CancellationToken to
             Trace.Info($"Current running agent version is {BuildConstants.AgentPackage.Version}");
             PackageVersion agentVersion = new PackageVersion(BuildConstants.AgentPackage.Version);
 
-            return serverVersion.CompareTo(agentVersion) > 0;
+            if (serverVersion.CompareTo(agentVersion) > 0)
+            {
+                return true;
+            }
+
+            if (Boolean.TryParse(Environment.GetEnvironmentVariable("AZP_AGENT_DOWNGRADE_DISABLED"), out var downgradeDisabled) && downgradeDisabled)
```

Comment:
Can we make this a Knob now?

Reply:
We need an execution context for that, right?

Reply:
we need a ```IKnobValueContext``` which we could create a custom implementation of that for SelfUpdater

Reply:
Oh, I see - updated!